### DISCUSSION TITLE
Conflict with substitude mode

### DISCUSSION THREAD

**Discussion by leana8959** [ggandor/leap.nvim#153] [Q&A] [answered]
Hello,

When I hit `s`, I seem to enter the substitute mode.\
I tried to remove the built-in `s` mapping by mapping it to something else, or simply map it to `<nop>`. However, it didn't stop removing the next character when I hit `s`.\
This is what I tried:
```lua
vim.keymap.set("n", "s", "<nop>", { desc = "It's the worse place in the universe" })
```

I can't seem to find the conflicting mapping.\
Here's the output of `verbose map s`
```text
o  s           * <Plug>(leap-forward-to)
                 Leap forward to
        Last set from Lua
x  s           * <Plug>(leap-forward-to)
                 Leap forward to
        Last set from Lua
n  s<Right>    * :ISwapNodeWithRight<CR>
                 Swap argument with right
        Last set from Lua
n  s<Left>     * :ISwapNodeWithLeft<CR>
                 Swap argument with left
        Last set from Lua
```

I did call `require('leap').add_default_mappings()` in a file in the `after/plugin` directory to add the default mappings. In fact, `S` works (reverse search), it's just the `s` that's somehow conflicting with something that I can't understand.

Thank you!

**Comment by leana8959** [comment]
I noticed that it's conflicting with these bindings:
```lua
vim.keymap.set('n', "s<Left>", ":ISwapNodeWithLeft<CR>", { desc = "Swap argument with left" })
vim.keymap.set('n', "s<Right>", ":ISwapNodeWithRight<CR>", { desc = "Swap argument with right" })
```
What I don't understand is, after the delay (to distinguish between `s` and `s<Left>` for example) it falls back to substitute mode and not leap.

**Reply by IndianBoy42** [reply, marked as answer]
If that's the full output of `verbose map s` then there is no leap mapping for `s` in normal mode. Do you see it in `verbose map S`. your `s` to `<nop>` mapping is causing leap to not add a mapping for `s` (this is documented)

**Reply by leana8959** [reply]
I removed the conflicting keymaps and it worked.
I do see leap in `verbose map S`. 

I'm still not sure why neovim would fall back to substitute mode other than leap, but that's not really a leap problem, so I'm closing it now. Thanks for your help!

**Reply by IndianBoy42** [reply]
There was no leap mapping to fallback onto. Leap did not add it because it saw a mapping for s already. I don't think neovim nor leap did anything wrong or unexpected here

**Reply by leana8959** [reply]
I'm not sure that I understand, when I type "s" there was a delay because neovim has to determine whether it's only `"s"` (leap), or `"s<right>"` and `"s<left>"` (ISwap). When the delay ends, it should either trigger leap and not substitute mode? Could you elaborate why this is the case? Thanks

**Reply by IndianBoy42** [reply]
It's deciding between s (substitute) and s+lr (iswap). As you saw from nmap, there is no leap.

**Reply by leana8959** [reply]
Well, maybe I had an error somewhere before. I already have a working config, I'm simply still curious why that was the case. I'll guess we'll leave the previous problem here, thank you for your time.
